tools/install-monorepo.sh

```diff
@@ -15,143 +17,344 @@ function usage {
+			die "Unable to install PHP $PHP_VERSION!"
+		fi
+	fi
+	echo "PHP $PHP_VERSION: available"
+}
+
+function do_composer {
+	if ! has_command composer; then
+		echo "composer: not found"
+		echo "Installing composer..."
+
+		# Note that installing from brew adds various unneeded dependencies (e.g. extra PHP).
+		# Largely grabbed from official install script:
+		# https://getcomposer.org/doc/faqs/how-to-install-composer-programmatically.md
+		EXPECTED_CHECKSUM="$(php -r 'copy("https://composer.github.io/installer.sig", "php://stdout");')"
+		php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
+		ACTUAL_CHECKSUM="$(php -r "echo hash_file('sha384', 'composer-setup.php');")"
+
+		if [[ "$EXPECTED_CHECKSUM" != "$ACTUAL_CHECKSUM" ]]; then
+			rm composer-setup.php
+			die "Invalid composer installer checksum!"
+		fi
+
+		php composer-setup.php --quiet --version="$COMPOSER_VERSION"
+		RESULT=$?
 		rm composer-setup.php
-		die 'ERROR: Invalid composer installer checksum'
+		if [[ $RESULT -ne 0 ]]; then
+			die "Unable to install composer!"
+			if [[ -f composer.phar ]]; then
+				# Clean up.
+				rm composer.phar
+			fi
+		fi
+		sudo mkdir -p /usr/local/bin
+		sudo mv composer.phar /usr/local/bin/composer
+		if ! has_command composer; then
+			die "Unable to install composer!"
+		fi
+		echo "Installed composer."
+	fi
+	echo "composer: available"
+}
+
+function do_monorepo_config {
+	echo "Setting up the Jetpack monorepo tooling..."
+	pnpm install
+	composer install
+	pnpm jetpack cli link &>/dev/null
+	if ! command -v jetpack &>/dev/null; then
+		warn 'Failed to link Jetpack CLI. The `jetpack` command will be unavailable.'
+		warn 'You can still use `pnpm jetpack` or set up an alias as desired.'
 	fi
+}
 
-	php composer-setup.php --version=$COMPOSER_VERSION --quiet
-	RESULT=$?
-	rm composer-setup.php
-	sudo mkdir -p /usr/local/bin
-	sudo mv composer.phar /usr/local/bin/composer
+if [[ $1 ]]; then
+	usage
 fi
 
-# Setup the Jetpack CLI
-info "Setting up the Jetpack CLI"
-pnpm install
-pnpm jetpack cli link
-if ! command -v jetpack &>/dev/null; then
-	warn 'Failed to link Jetpack CLI. The `jetpack` command will be unavailable.'
-	warn 'Likely this is because `pnpm setup` has not been run successfully. You may use `pnpm jetpack` from within the monorepo checkout instead.'
-fi
-pnpm jetpack install --root
-
-success "Installation complete. You may need to restart your terminal for changes to take effect. Then you can run tools/check-development-environment.sh to make sure everything installed correctly."
-
-# Reset the terminal so it picks up the changes.
-if [[ "$SHELL" == "/bin/zsh" ]]; then
-	info "Refreshing terminal"
-	exec zsh
-elif [[ "$SHELL" == "/bin/bash" ]]; then
-	info "Refreshing terminal"
-	exec bash
-else
-	info "Note: You may have to restart your terminal for monorepo tools to work properly."
-fi
+do_system
+do_basics
+do_homebrew
```

Comment:
I tried this on my existing system so it had nothing to do, but there was a decent delay between the output of `do_basics` and `do_homebrew` and another decent wait for `do_nvm`. Everything else felt "instant". Not sure if either of the commands could check faster (again in this case, they were already installed at sufficient versions) or if we can let folks know what's happening and that the script didn't hang. 

I could see this being as simple as an `info "Checking OS"`, `info "Checking Git and Curl"`, `info Checking for Homebrew`, etc. I like the UX of an indicator but I am not married to that idea at all.

Reply:
I'll find an old laptop, wipe it, and try it on a fresh fresh system (at least at whatever Mac OS version the old laptop supports) tonight or tomorrow.

Reply:
> there was a decent delay between the output of `do_basics` and `do_homebrew`
> Not sure if either of the commands could check faster

Yes, this is because it runs `brew update` silently, but I'll look into making a few things more verbose and perhaps move the `brew update` to only run when installing.

> another decent wait for `do_nvm`

This one's a bit odd...it grabs a GH release URL, but that should be pretty quick in most cases. Perhaps a hiccup? Does it still do this?